File: CppChatClient/message/TipNotificationContent.cpp
//
//  TipNotificationContent.cpp
//  ChatClient
//
//  Created by Hao Jia on 2020/1/31.
//  Copyright © 2020 WF Chat. All rights reserved.
//

#include "TipNotificationContent.h"

namespace WFCLib {

TipNotificationContentPrototype TipNotificationContent::sPrototype;

MessagePayload TipNotificationContent::encode() const {
    MessagePayload payload;
    payload.extra = extra;
    payload.content = tip;
    return payload;
}

void TipNotificationContent::decode(const MessagePayload & payload) {
    extra = payload.extra;
    tip = payload.content;
}


MessageContent* TipNotificationContent::clone() const {
    TipNotificationContent *p = new TipNotificationContent();
    p->extra = extra;
    p->tip = tip;
    return p;
}

const MessageContentPrototype* TipNotificationContent::getPrototype() const {
    return &sPrototype;
}

std::string TipNotificationContent::digest() const {
    return tip;
}

std::string TipNotificationContent::formatNotification(const Message &msg) const {
    return tip;
}
};

File: CsChatClient/Messages/Notification/TipNotificationContent.cs
using System;

namespace CsChatClient.Messages.Notification
{
    [ContentAttribute(MessageContentType.MESSAGE_CONTENT_TYPE_TIP, MessageContentPersistFlag.PersistFlag_PERSIST)]
    public class TipNotificationContent : NotificationMessageContent
    {
        public string Tip { get; set; }
        public override void Decode(MessagePayload payload)
        {
            Tip = payload.Content;
        }

        public override string Digest(MessageEx message)
        {
            return Tip;
        }

        public override MessagePayload Encode()
        {
            MessagePayload payload = new MessagePayload();
            payload.Content = Tip;
            return payload;
        }

        public override string FormatNotification(MessageEx message)
        {
            return Digest(message);
        }
    }
}
